include/base_interface.h

```diff
@@ -0,0 +1,322 @@
+   */
+  virtual ~BaseInterface() {};
+
+  /**
+   * Constructor. It takes the name of the subsection within the
+   * parameter file, the finite element used to discretize the system,
+   * the name of the components and a string were the block of
+   * differential and algebraic components are specified.
+   */
+  BaseInterface(const std::string &name="",
+                const std::string &default_fe="FE_Q(1)",
+                const std::string &default_component_names="u",
+                const std::string &default_differential_components="");
+
+  void init ();
+
+  virtual void declare_parameters (ParameterHandler &prm);
+
+  virtual void parse_parameters_call_back ();
+  /**
+   * returns the vector @p of differential blocks
+   */
+  const std::vector<unsigned int> get_differential_blocks() const;
+
+  /**
+   * set time to @p t for forcing terms and boundary conditions
+   */
+  virtual void set_time (const double &t) const;
+
+
+  /**
+   * This function is used to modify triangulation using boundary_id or manifold_id.
+   * In the case a signal is required, this is the function to modify.
+   */
+  virtual void postprocess_newly_created_triangulation(Triangulation<dim, spacedim> &tria) const;
+
+  /**
+   * Applies Dirichlet boundary conditions
+   *
+   * This function is used to applies Dirichlet boundary conditions.
+   * It takes as argument a DoF handler @p dof_handler and a constraint
+   * matrix @p constraints.
+   *
+   */
+  void apply_dirichlet_bcs (const DoFHandler<dim,spacedim> &dof_handler,
+                            ConstraintMatrix &constraints) const;
+
+  /**
+   * Applies Neumann boundary conditions
+   *
+   */
+  void apply_neumann_bcs (const typename DoFHandler<dim,spacedim>::active_cell_iterator &cell,
+                          FEValuesCache<dim,spacedim> &scratch,
+                          std::vector<double> &local_residual) const;
+
+
+  /**
+   * Applies CONSERVATIVE forcing terms.
+   * This function applies the conservative forcing terms, which can be
+   * defined by expressions in the parameter file.
+   *
+   * If the problem involves NON-conservative loads, they must be included
+   * in the residual formulation.
+   *
+   */
+  void apply_forcing_terms (const typename DoFHandler<dim,spacedim>::active_cell_iterator &cell,
+                            FEValuesCache<dim,spacedim> &scratch,
+                            std::vector<double> &local_residual) const;
+
+  /**
+   * Initialize all data required for the system
+   *
+   * This function is used to initialize the varibale AnyData @p d
+   * that contains all data of the problem (solutions, DoF, quadrature
+   * points, solutions vector, etc ).
+   * It takes as argument the number of DoF per cell @p dofs_per_cell,
+   * the number of quadrature points @p n_q_points, the number of
+   * quadrature points per face @p n_face_q_points, the reference to
+   * solutions vectors @p sol and the reference to the AnyData @p d.
```

Comment:
This comment should be updated. n_q_points, n_face_q_points, etc. should all disappear from the comment